fenix/app/src/main/java/org/mozilla/fenix/components/metrics/clientdeduplication/ClientDeduplicationPing.kt

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics.clientdeduplication
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.GleanMetrics.Activation
+import org.mozilla.fenix.GleanMetrics.ClientDeduplication
+import org.mozilla.fenix.GleanMetrics.Pings
+import org.mozilla.fenix.components.metrics.MetricsUtils.getHashedIdentifier
+
+/**
+ * Class to help construct and send the `clientDeduplication` ping.
+ */
+class ClientDeduplicationPing(private val context: Context) {
+    private val customHashingSalt = "bug-1813195-02-2023"
+
+    /**
+     * Fills the metrics and triggers the 'clientDeduplication' ping.
+     */
+    internal fun triggerPing(active: Boolean) {
+        CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Worth pointing out that in theory you can have concurrent runs of this code e.g., if the user goes through start/stop cycles really quickly, we will `launch` multiple coroutines here that can run in any order and concurrently, because Dispatchers.IO is backed by multiple threads. Looks like `Activation.identifier.set` and `ClientDeduplication.validAdvertisingId.set`  etc. are all shared state and could be written to concurrently.

I don't think this will really negatively affect your results here though, but if we're worried (and happy to talk through more) then we can use a custom CoroutineScope and dispatcher, backed by a single thread e.g., https://github.com/mozilla-mobile/firefox-android/blob/a05ece3df1f60e7b10ab4b564b82f47b7ba8d0eb/fenix/app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataService.kt#L56-L59

Reply:
Apologies, I did miss this!

As long as this shouldn't impact our new pings, then I can revisit this later. I will bring this up with my team and we can create a bug to address this in the future. If we weren't trying to get this PR landed soon I would take this now, but sounds like something we can tackle later and fleshed out so none of our pings run into this issue.

@travis79 @Dexterp37